router/batchrouter/asyncdestinationmanager/yandex-metrica-offline-events/yandexmetricaofflineevents.go

```diff
@@ -0,0 +1,161 @@
+	YandexMetricaBulkUpload := &YandexMetricaBulkUploader{
+		destName:          destination.DestinationDefinition.Name,
+		destinationConfig: destination.Config,
+		pollUrl:           "",
+		transformUrl:      config.GetString("DEST_TRANSFORM_URL", "http://localhost:9090"),
+		logger:            logger.NewLogger().Child("batchRouter").Child("AsyncDestinationManager").Child("YandexMetrica").Child("YandexMetricaBulkUploader"),
+		timeout:           config.GetDuration("HttpClient.yandexMetricaBulkUpload.timeout", 30, time.Second),
+	}
+	return YandexMetricaBulkUpload, nil
+}
+
+// return a success response for the poll request every time by default
+func (ym *YandexMetricaBulkUploader) Poll(pollInput common.AsyncPoll) common.PollStatusResponse {
+	return common.PollStatusResponse{
+		Complete:       true,
+		InProgress:     false,
+		StatusCode:     200,
+		HasFailed:      false,
+		HasWarning:     false,
+		FailedJobURLs:  "",
+		WarningJobURLs: "",
+		Error:          "",
+	}
+}
+
+type YandexMetricaOfflineEvents struct {
+	// Add your configuration fields here
+}
+
+func NewYandexMetricaOfflineEvents() *YandexMetricaOfflineEvents {
+	// Initialize and return a new instance of YandexMetricaOfflineEvents
+	return &YandexMetricaOfflineEvents{}
+}
+
+func (ym *YandexMetricaBulkUploader) GetUploadStats(UploadStatsInput common.GetUploadStatsInput) common.GetUploadStatsResponse {
+	return common.GetUploadStatsResponse{}
+}
+
+func (ym *YandexMetricaBulkUploader) Upload(asyncDestStruct *common.AsyncDestinationStruct) common.AsyncUploadOutput {
+	destination := asyncDestStruct.Destination
+	destinationID := destination.ID
+	filePath := asyncDestStruct.FileName
+	destConfig := destination.Config
+	destType := destination.DestinationDefinition.Name
+	failedJobIDs := asyncDestStruct.FailedJobIDs
+	importingJobIDs := asyncDestStruct.ImportingJobIDs
+
+	destinationUploadUrl := asyncDestStruct.DestinationUploadURL
+	uploadURL, err := url.JoinPath(ym.transformUrl, destinationUploadUrl)
+
+	if err != nil {
+		return common.AsyncUploadOutput{
+			FailedReason:  "BRT: Failed to prepare upload url " + err.Error(),
+			FailedCount:   len(failedJobIDs) + len(importingJobIDs),
+			DestinationID: destinationID,
+		}
+	}
+	file, err := os.Open(filePath)
+	if err != nil {
+		return common.AsyncUploadOutput{
+			FailedReason:  "Error while opening file",
+			DestinationID: destinationID,
+		}
+	}
+	defer file.Close()
+	var input []common.AsyncJob
+	scanner := bufio.NewScanner(file)
+
+	for scanner.Scan() {
+		var tempJob common.AsyncJob
+		jobBytes := scanner.Bytes()
+		err := json.Unmarshal(jobBytes, &tempJob)
+		if err != nil {
+			return common.AsyncUploadOutput{
+				FailedJobIDs:  append(failedJobIDs, importingJobIDs...),
+				FailedReason:  "BRT: Error in Unmarshalling Job for Yandex Metrica destination: " + err.Error(),
+				FailedCount:   len(failedJobIDs) + len(importingJobIDs),
+				DestinationID: destinationID,
+			}
+		}
+		input = append(input, tempJob)
+	}
```

Comment:
Alternatively using JSON decoder 

https://abhinavg.net/2020/07/02/parse-ndjson/

```suggestion
	file, err := os.Open(filePath)
	if err != nil {
		return common.AsyncUploadOutput{
			FailedReason:  "Error while opening file",
			DestinationID: destinationID,
		}
	}
	defer file.Close()
	decoder := json.NewDecoder(file)
	var input []common.AsyncJob

	for decoder.More() {
	    var tempJob common.AsyncJob
		err := decoder.Decode(&tempJob)
		if err != nil {
			return common.AsyncUploadOutput{
				FailedJobIDs:  append(failedJobIDs, importingJobIDs...),
				FailedReason:  "BRT: Error in Unmarshalling Job for Yandex Metrica destination: " + err.Error(),
				FailedCount:   len(failedJobIDs) + len(importingJobIDs),
				DestinationID: destinationID,
			}
		}
		input = append(input, tempJob)
	}
```